### DISCUSSION TITLE
Computed Field that cannot be inputted but generated dynamically?

### DISCUSSION THREAD

**Discussion by v3ss0n** [pydantic/pydantic#4677] [Question]
How can i have a field that is not available for input but only generated dynamically?

```python

class Bamm(BaseModel):
    f1 : str
    f2 : str
    f3 : str = ""
    @validator("f3",check_fields=False,always=True,pre=True)
    def gen_f3(cls,v,values):
        return f"{values['f1']} and {values['f2']}"
c = Bamm(**dict(f1="hello",f2="world"))
print (c.dict())

```
Here is my model.
I would like to generate f3 , but now f3 can be defined from input.
i want a filed that cannot be changed from input but only computed based on others.

**Comment by PrettyWood** [comment]
what you did is probably the best you can do for now. It will be possible in v2 I promise (see https://github.com/pydantic/pydantic/issues/935)
